src/corefx/System.Globalization.Native/collation.cpp

```diff
@@ -0,0 +1,254 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include <assert.h>
+#include <stdint.h>
+#include <unicode/uchar.h>
+#include <unicode/ucol.h>
+#include <unicode/usearch.h>
+#include <unicode/utf16.h>
+
+const int32_t CompareOptionsIgnoreCase = 1;
+// const int32_t CompareOptionsIgnoreNonSpace = 2;
+// const int32_t CompareOptionsIgnoreSymbols = 4;
+// const int32_t CompareOptionsIgnoreKanaType = 8;
+// const int32_t CompareOptionsIgnoreWidth = 0x10;
+// const int32_t CompareOptionsStringSort = 0x20000000;
+
+/*
+ * To collator returned by this function is owned by the callee and must be closed when this method returns
+ * with a U_SUCCESS UErrorCode.
+ *
+ * On error, the return value is undefined.
+ */
+UCollator* GetCollatorForLocaleAndOptions(const char* lpLocaleName, int32_t options, UErrorCode* pErr)
+{
+    UCollator* pColl = nullptr;
+
+    pColl = ucol_open(lpLocaleName, pErr);
+
+    if ((options & CompareOptionsIgnoreCase) == CompareOptionsIgnoreCase)
```

Comment:
Can ucol_open fail, and if so, should we not call ucol_setAttribute if it does?  Or does ICU chain these such that ucol_setAttribute will check for a previous error passed in and immediately return that same error if there was one?